eng/source-build-patches/0012-Build-Roslyn-as-net6.0-in-source-build.patch

```diff
@@ -0,0 +1,305 @@
+From ed6e61977542a2fb1f69da41eb9806a4212d4046 Mon Sep 17 00:00:00 2001
+From: Davis Goodin <dagood@microsoft.com>
+Date: Mon, 28 Dec 2020 18:21:27 -0600
+Subject: [PATCH] Build Roslyn as net6.0 in source-build
```

Comment:
@jaredpar this patch may end up being a bit troublesome. Not sure what you think.

Reply:
Yeah this patch is problematic. It's essentially forcing us to support running on `netcoreapp3.1` even though our code is developed with the assumption that this is just not possible. 

What is the reasoning behind this? If we actually need to support `netcoreapp3.1` then we may have to accept that. But even so I'd just add it back into the main tree. As long as it exists as a patch like this we're just going to be breaking support left and right here. It's not sustainable.